### PR TITLE
cli/zip: escape job description newlines in crdb_internal.jobs.txt

### LINKED ISSUE #110349: jobs: escape job description newlines in crdb_internal.jobs.txt
The row level TTL job generates multiple lines per job in `crdb_internal.jobs.txt`:
```
894609672582266883      ROW LEVEL TTL   "ttl for defaultdb.public.failed_jobs
-- for each span, iterate to find rows:
SELECT id FROM [190 AS tbl_name]
AS OF SYSTEM TIME INTERVAL '-30 seconds'
WHERE <crdb_internal_expiration OR ttl_expiration_expression> <= $1
AND (id) > (<span start>) AND (id) < (<span end>)
ORDER BY id
LIMIT <ttl_select_batch_size>
-- then delete with:
DELETE FROM [190 AS tbl_name]
WHERE <crdb_internal_expiration OR ttl_expiration_expression> <= $1
AND (id) IN (<rows selected above>)"   
```

This causes issues with commands like `grep -v succeeded crdb_internal.jobs.txt`.

To fix this, escape newlines with `\n` in the output.

Thread:
https://cockroachlabs.slack.com/archives/C0168LW5THS/p1694442917840589?thread_ts=1694439446.347799&cid=C0168LW5THS

See also:
https://github.com/cockroachlabs/support/issues/2593#issuecomment-1714127383

Jira issue: CRDB-31391

### PR DESCRIPTION
**Pull request by ecwall**
Fixes #110349

The row level TTL job generates multiple lines per job in crdb_internal.jobs.txt:
```
894609672582266883      ROW LEVEL TTL   "ttl for defaultdb.public.failed_jobs
-- for each span, iterate to find rows:
SELECT id FROM [190 AS tbl_name]
AS OF SYSTEM TIME INTERVAL '-30 seconds'
WHERE <crdb_internal_expiration OR ttl_expiration_expression> <= $1
AND (id) > (<span start>) AND (id) < (<span end>)
ORDER BY id
LIMIT <ttl_select_batch_size>
-- then delete with:
DELETE FROM [190 AS tbl_name]
WHERE <crdb_internal_expiration OR ttl_expiration_expression> <= $1
AND (id) IN (<rows selected above>)"
```
This causes issues with commands like grep -v succeeded crdb_internal.jobs.txt.

This PR fixes this by escaping newlines with \n in the output.

Release note: None

### CODE REVIEW

File: pkg/cli/clisqlexec/run_query.go [hunk 1]
```diff
@@ -42,7 +42,11 @@ func (sqlExecCtx *Context) RunQuery(
 // Timings is enabled are written to 'tw'.
 // Errors and warnings, if any, are printed to 'ew'.
 func (sqlExecCtx *Context) RunQueryAndFormatResults(
-	ctx context.Context, conn clisqlclient.Conn, w, tw, ew io.Writer, fn clisqlclient.QueryFn,
+	ctx context.Context,
+	conn clisqlclient.Conn,
+	w, tw, ew io.Writer,
+	fn clisqlclient.QueryFn,
+	escapeNewline bool,
```
Comment (knz) [inline pkg/cli/clisqlexec/run_query.go:49]: This can be a new field on the Context struct.

File: pkg/cli/zip.go [hunk 2]
```diff
@@ -428,7 +428,15 @@ func (zc *debugZipContext) dumpTableDataForZip(
...
 			// in-RAM buffering.
-			return sqlExecCtx.RunQueryAndFormatResults(ctx, conn, w, io.Discard, stderr, clisqlclient.MakeQuery(query))
+			return sqlExecCtx.RunQueryAndFormatResults(
+				ctx,
+				conn,
+				w,
+				io.Discard,
+				stderr,
+				clisqlclient.MakeQuery(query),
+				true, /* escapeNewline */
```
Comment (knz) [inline pkg/cli/zip.go:438]: I don't think it's reasonable to set this unconditionally.

The zip command supports a command line flag to change the format of the retrieved files IIRC. With Json for example this escaping would be incorrect.

**Review by knz — COMMENTED** [review]
I would prefer if this was changing fewer function call interfaces. I made a suggestion to that effect below.

File: pkg/cli/clisqlexec/run_query.go (hunk 1 above)
Comment (ecwall) [reply pkg/cli/clisqlexec/run_query.go:49]: `cli.sqlExecCtx` is used in many places so this field can't be moved there unless we either
1) make a new instance for use by zip.go
2) want to affect all places using it

File: pkg/cli/zip.go (hunk 2 above)
Comment (ecwall) [reply pkg/cli/zip.go:438]: I changed this to only affect `csvReporter`.

**Review by rafiss — APPROVED** [review]

**Review by rafiss — CHANGES_REQUESTED** [review]
actually - wanted to see if there's an answer to Raphael's suggestion:

> Make a new field in `sqlExecCtx` that defaults to false, and then set it in zipCmd.Run in cli/zip.go.

**Pull request closed without merging**